### DISCUSSION TITLE
I can't set webhook

### DISCUSSION THREAD

**Discussion by abdxm** [def-studio/telegraph#320] [Q&A]
DefStudio\Telegraph\Exceptions\TelegramWebhookException 

  You application must have a secure (https) url in order to accept webhook calls

  at vendor/defstudio/telegraph/src/Exceptions/TelegramWebhookException.php:24
     20▕     }
     21▕ 
     22▕     public static function invalidScheme(): TelegramWebhookException
     23▕     {
  ➜  24▕         return new self("You application must have a secure (https) url in order to accept webhook calls");
     25▕     }
     26▕ }
     27▕ 

      +17 vendor frames 
  18  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()

**Comment by fabio-ivona** [comment]
Hi, you must have a valid https url, otherwise telegram doesn't accept your webhook url

**Reply by esmaei1** [reply]
I have a valid https url but get same error.

**Reply by fabio-ivona** [reply]
@esmaei1 the generated url from this function: `route('telegraph.webhook', 1)` should return an URL that starts with `https://`

could you check it?

**Comment by amirmsj1374** [comment]
Hi I have same problem and checked the route('telegraph.webhook', 1) but and it returns back my ngrok URL includes 'https://'

**Reply by fabio-ivona** [reply]
could you paste here a stacktrace?

**Comment by MacroTom** [comment]
Use ngrok. Create an account. And make sure to add the https url given to you, in your APP_URL environment variable.
